scripts/utils.py

```diff
@@ -0,0 +1,178 @@
+import base64
+import re
+import argparse
+import boto3
+
+# def Base64ToHexHash(base64_hash):
+#    return hexlify(base64.decodestring(base64_hash.strip('\n"\'')))
+
+
+def get_bucket_name(fi, PROJECT_MAP):
+    """
+    TCGA-open -> gdc-tcga-phs000178-open
+    TCGA-controled -> gdc-tcga-phs000178-controled
+    TARGET-controled -> gdc-target-phs000218-controlled
+    """
+    bucketname = ""
+    project = fi.get("project_id", "").split("-")[0]
+    if fi.get("acl", "") == "*" or fi.get("acl", "") == "['open']":
+        bucketname = "gdc-" + PROJECT_MAP.get(project, "") + "-open"
+    else:
+        bucketname = "gdc-" + PROJECT_MAP.get(project, "") + "-controlled"
+    return bucketname
+
+
+def extract_md5_from_text(text):
+    """
+    extract base-128 hash from string
+    Args:
+        text(str): string output from gsutil command
+    Returns:
+        hash(str) or None
+    """
+    m = re.search("[0-9,a-f]{32}", text.lower())
+    if m:
+        return m.group(0)
+    else:
+        return None
+
+
+def get_headers(manifest_file):
```

Comment:
missing docstring

Reply:
fixed in d47a36c889171c98bbe27a87e89fb6d55d356f0f